test/xpu_api/utilities/function.objects/refwrap/weak_result.pass.cpp

```diff
@@ -0,0 +1,143 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/functional>
+#include <oneapi/dpl/type_traits>
+
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+class KernelWeakResultTest;
+
+template <class Arg, class Result>
+struct my_unary_function
+{ // std::unary_function was removed in C++17
+    typedef Arg argument_type;
+    typedef Result result_type;
+};
+
+template <class Arg1, class Arg2, class Result>
+struct my_binary_function
+{ // std::binary_function was removed in C++17
+    typedef Arg1 first_argument_type;
+    typedef Arg2 second_argument_type;
+    typedef Result result_type;
+};
+
+class functor1 : public my_unary_function<int, char>
+{
+};
+
+class functor2 : public my_binary_function<char, int, float>
+{
+};
+
+class functor3 : public my_unary_function<char, int>, public my_binary_function<char, int, float>
+{
+  public:
+    typedef float result_type;
+};
+
+class functor4 : public my_unary_function<char, int>, public my_binary_function<char, int, float>
+{
+  public:
+};
+
+class C
+{
+};
+
+template <class T>
+struct has_result_type
+{
+  private:
+    struct two
+    {
+        char _;
+        char __;
+    };
+    template <class U>
+    static two
```

Comment:
`two` seems unnecessary. `int` can be used instead.

Reply:
Fixed